tests/core/test_accounts.py

```diff
@@ -428,48 +428,74 @@ def encrypt_mark_params():
     public_key = '0x4c0883a69102937d6231471b5dbb6204fe5129617082792ae468d01a3f362318'
     password = 'test!'
 
-    # 'private_key, password, kdf, expected_decrypted_key, expected_kdf'
+    # 'private_key, password, kdf, iterations, expected_decrypted_key, expected_kdf'
     return [
         (
             public_key,
             password,
             None,
+            None,
             to_bytes(hexstr=public_key),
             'scrypt'
         ),
         (
             public_key,
             password,
             'pbkdf2',
+            None,
             to_bytes(hexstr=public_key),
             'pbkdf2'
         ),
         (
             keys.PrivateKey(HexBytes(public_key)),
             password,
             None,
+            None,
             keys.PrivateKey(HexBytes(public_key)).to_bytes(),
             'scrypt'
         ),
+        (
+            public_key,
+            password,
+            'pbkdf2',
+            1024,
+            to_bytes(hexstr=public_key),
+            'pbkdf2'
+        ),
+        (
+            public_key,
+            password,
+            'scrypt',
+            1024,
```

Comment:
At least two things would improve these tests:

1. run an example where `kdf` is `None` and `iterations` is set.
2. verify the iterations value in the keyfile (note that the iterations value output is sensitive to which kdf is used)

Reply:
Rearranged the tests a bit. First and second tests are just the basic private key ones, the rest are messing with the KDF and iterations. The fourth test (new one I added) is the one you're asking about:
* https://github.com/ethereum/eth-account/pull/52/commits/aa9182811a67a673dc428f1f37ea41fad14754fc#diff-1de692d823115945c2232e1aa3066811R463
* https://github.com/ethereum/eth-account/pull/52/commits/aa9182811a67a673dc428f1f37ea41fad14754fc#diff-1de692d823115945c2232e1aa3066811R540

And added the iterations validation [here](https://github.com/ethereum/eth-account/pull/52/commits/aa9182811a67a673dc428f1f37ea41fad14754fc#diff-1de692d823115945c2232e1aa3066811R518) and [here](https://github.com/ethereum/eth-account/pull/52/commits/aa9182811a67a673dc428f1f37ea41fad14754fc#diff-1de692d823115945c2232e1aa3066811R564).